splink/blocking.py

```diff
@@ -24,13 +25,14 @@ def blocking_rule_to_obj(br):
         if blocking_rule is None:
             raise ValueError("No blocking rule submitted...")
         sqlglot_dialect = br.get("sql_dialect", None)
-        salting_partitions = br.get("salting_partitions", 1)
 
-        return BlockingRule(
-            blocking_rule,
-            salting_partitions,
-            sqlglot_dialect,
-        )
+        salting_partitions = br.get("salting_partitions", None)
+        if salting_partitions is None:
+            return BlockingRule(blocking_rule, sqlglot_dialect)
+        else:
+            return SaltedBlockingRule(
+                blocking_rule, sqlglot_dialect, salting_partitions
+            )
```

Comment:
We should probably clean this up at a later date.

The if else blocks are difficult to read in parts.

Reply:
(this was my doing)

Reply:
Yeah, agreed.  Should be named properly (called a factory function) and the logic simplified.  Let's do that separately

Reply:
We could potentially use an @classmethod for dictionaries that are passed in.

See here for an example of what I mean - https://github.com/moj-analytical-services/lazy_comparison_brs/blob/comparison_level_demo/comparison_level.py#L39.

Certainly not for this PR, it's merely something to ruminate on.

Reply:
(this would be part of a wider function factory which would eliminate a lot of these if statements)